### PR TITLE
[Hub Generated] Review request for Creator to add version preview/2022-09-01-preview

### PR DESCRIPTION
**Pull request by zhugexubin**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/zhugexubin/azure-rest-api-specs/dev-maps-Creator-2022-09-01-preview...main/).
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*engagement record*](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/271/Azure-REST-API-Stewardship?anchor=rest-api-stewardship-process).

* Link to engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous Open API document (swagger) if applicable, and the root paths that have been updated. </sup>
* Design Document:
* Previous Open API Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azapi) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azapi).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

**Comment by zhugexubin** [comment]
@tjprescott  Hi Travis, Can you take a look at this PR. The swagger doc detects BreakingChange, but there is no breaking change for this PR since this swagger doc has not yet been published and not visible to public for now. We made some additional changes on the swagger doc to improve the content and correctness.

**Comment by tjprescott** [comment]
@mikekistler or @tg-msft could either of you review this from the perspective of the API Stewardship board? This PR introduces deviations from what the board had previously signed off on and the tooling is flagging numerous issues.

File: specification/maps/data-plane/Creator/preview/2022-09-01-preview/mapconfiguration.json [hunk 1]
```diff
@@ -117,6 +117,13 @@
           {
             "$ref": "../../../Common/preview/1.0/common.json#/parameters/ClientId"
           },
+          {
+            "name": "Content-Type",
+            "description": "Specify the request body content type. `application/zip` if request is a zip file, `application/json` if request body is a JSON file.",
+            "type": "string",
+            "in": "header",
+            "required": true
+          },
```
Comment (mikekistler) [inline specification/maps/data-plane/Creator/preview/2022-09-01-preview/mapconfiguration.json:126]: You should not specify the `Content-Type` header explicitly. Use `consumes`.

File: specification/maps/data-plane/Creator/preview/2022-09-01-preview/mapconfiguration.json [hunk 2]
```diff
@@ -350,44 +360,66 @@
...
         },
         "description": {
-          "description": "The description for the style. Maximum length 1024 characters.",
+          "description": "The description for the style. Maximum length 500 characters.",
           "type": "string"
         },
         "theme": {
-          "description": "The theme for client SDK widgets(style picker, zoom, level).",
-          "type": "string"
+          "description": "The theme for client SDK widgets(style picker, zoom, level picker). Maximum length 100 characters.",
```
Comment (mikekistler) [inline specification/maps/data-plane/Creator/preview/2022-09-01-preview/mapconfiguration.json:396]: This description doesn't seem to match the enum values defined. Also, it makes little sense to say "Maximum length 100 characters" when the largest enum value is only 5 characters.

File: specification/maps/data-plane/Creator/preview/2022-09-01-preview/style.json [hunk 3]
```diff
@@ -102,21 +102,28 @@
...
           {
             "$ref": "../../../Common/preview/1.0/common.json#/parameters/ClientId"
           },
+          {
+            "name": "Content-Type",
+            "description": "Specify the request body content type. `application/zip` if request is a zip file, `application/json` if request body is a JSON file.",
+            "type": "string",
+            "in": "header",
+            "required": true
+          },
```
Comment (mikekistler) [inline specification/maps/data-plane/Creator/preview/2022-09-01-preview/style.json:126]: Don't specify `Content-Type` explicitly -- use `consumes`.

**Review by mikekistler — APPROVED** [review]
This mostly looks fine to me -- mostly changes to descriptions and better documentation is always welcome.

I did point out a few minor concerns that I hope will be addressed.

It looks like the breaking changes are only breaking in relation to a prior preview version -- if so that should be fine.  But we generally require _any_ changes to be done in a new API version, but this too can be relaxed if the API is not yet in public preview. Please bring this to the breaking changes office hours to close on those details.

File: specification/maps/data-plane/Creator/preview/2022-09-01-preview/mapconfiguration.json (hunk 2 above)
Comment (zhugexubin) [reply specification/maps/data-plane/Creator/preview/2022-09-01-preview/mapconfiguration.json]: Updated the description and removed the 100 characters length.

File: specification/maps/data-plane/Creator/preview/2022-09-01-preview/style.json (hunk 3 above)
Comment (zhugexubin) [reply specification/maps/data-plane/Creator/preview/2022-09-01-preview/style.json]: Switched to use consumes for Content-Type

File: specification/maps/data-plane/Creator/preview/2022-09-01-preview/mapconfiguration.json (hunk 1 above)
Comment (zhugexubin) [reply specification/maps/data-plane/Creator/preview/2022-09-01-preview/mapconfiguration.json]: Switched to use consumes

**Comment by zhugexubin** [comment]
> This mostly looks fine to me -- mostly changes to descriptions and better documentation is always welcome.
> 
> I did point out a few minor concerns that I hope will be addressed.
> 
> It looks like the breaking changes are only breaking in relation to a prior preview version -- if so that should be fine. But we generally require _any_ changes to be done in a new API version, but this too can be relaxed if the API is not yet in public preview. Please bring this to the breaking changes office hours to close on those details.

@mikekistler Thanks Mike for your comments, I resolved them with a new update. Are you suggesting we still need to attend breaking change office hours to close these even with your approval?

**Comment by mikekistler** [comment]
Confirmed offline with @senthilbing that this version of the API is not yet available to customers so no new api-version needed. This is good to go.

**Pull request merged**